### DISCUSSION TITLE
Donwload Thumbnail with a specific DPI

### DISCUSSION THREAD

**Discussion by IMCtoys** [pimcore/pimcore#15904] [Q&A / Support] [answered]
Hi, we would need to be able to choose via thumbnail whether to download the file in png or jpg but in 72dpi. I saw there is a Web optimized option that downloads the file at 72dpi, but choses the format (jpg or png) automatically. The question is: is it possible to create a thumbnail configuration for a specific DPI? Or, is there any alternative to be able to give the option to download the image in a specific format and specific resolution, as we can do in backend? thanks!

**Comment by NiklasBr** [comment, marked as answer]
DPI (and related PPI) is whatever you want, it is dependent upon how *you* render the item.

If you render a 300x300 pixel image on a 1x1 inch postage stamp, then you have 300 DPI.
If you render the 300x300 pixel image as 1:1 pixel to point image on an iPhone 14, it will be 460 DPI.
If you render the 300x300 image on Dell UltraSharp 27 4K PremierColor Monitor - UP2720Q, it will be 163 DPI.

Some images may hold certain metadata to hint to printers what the default DPI it should be printed at, but other than that you get what you want.

If you want to print a 72 DPI image at 3 inches in size you make it 3*72= 216 pixels.

**Reply by IMCtoys** [reply]
Hi Niklas, and thanks for your quick answer.

the idea is to use thumbnail to download the image in the portal, not to show the image in screen. We, in our current DAM (not pimcore) we set different download options modifying the size of the image in px, and the resolution in DPI. For example, some customers may want to download the image in 72dpi, 1500x1500 px and in png, other in 300dpi, 2000x2000 px in jpg, etc. Our images are upload always in 300dpi, so for that is not a problem, but we need to have the option to download in 71dpi and be able to choose into jpg and png. The option "web optimized" downloads the image at 72dpi, what is perfect, but we cannot choose the format (jpg or png). We know imagemagick can set the resolution, since we have used it for other projects, but we don't see how to made it work with pimcore.

Thanks!

**Reply by NiklasBr** [reply]
_DPI has nothing to do with image resolution, it is just a function of how it is rendered/displayed_

For your 72 DPI image at 1500x1500px you create one image size of 1500x1500 pixels, and do the same with the 300 DPI at 2000x2000 pixels: A thumbnail size of 2000x2000px.

To be able to properly calculate the DPI and PPI you must know exactly the physical size the image is being rendered at. The density/dpi value which may exist in the metadata of the image does not matter. You can add a DPI value och 9000 to a 100x100 image and it will not be different compared to a DPI of 7 on a 100x100 image. It's the exact same image, same size, same bits and bytes.

**Reply by NiklasBr** [reply]
Here's an example, first some info about the image:

```sh
$ ll testimage.png 
-rw-r--r-- 1 root root 1948071 Aug 21 14:35 testimage.png
$ file testimage.png 
testimage.png: PNG image data, 1326 x 820, 8-bit/color RGBA, non-interlaced
```

Then we can add metadata, which does nothing to the file dimensions.
```sh
$ exiftool testimage.png -xresolution=300 -v2
Writing JFIF:XResolution if tag exists
Writing Photoshop:XResolution if tag exists
Writing XMP-tiff:XResolution if tag exists
Writing IFD0:XResolution
======== testimage.png
Rewriting testimage.png...
  Editing tags in: IFD0 JFIF PNG Photoshop XMP 
  Creating tags in: IFD0 PNG 
  FileType = PNG
  FileTypeExtension = PNG
  MIMEType = image/png
PNG IHDR (13 bytes):
PNG iCCP (2729 bytes):
  + [iCCP directory]
PNG eXIf (86 bytes):
  + [TIFF directory]
  Rewriting IFD0
    + IFD0:XResolution = '300'
  Rewriting ExifIFD
PNG iTXt (471 bytes):
  Rewriting XMP
PNG IDAT (119 chunks, total 1943276 bytes)
PNG IEND (end of image)
    1 image files updated
```

Info again:
```sh
$ file testimage.png 
testimage.png: PNG image data, 1326 x 820, 8-bit/color RGBA, non-interlaced
$ ll testimage.png
-rw-r--r-- 1 root root 1948091 Sep  6 12:43 testimage.png
```

It's all about the display/rendering/printing size, not pixel size.

**Reply by IMCtoys** [reply]
Hi Niklas and thank you very much for you help on it. I'm trying to diggest everythink since I'm not expert in graphics and all my previous imput from my graphic colleagues and customers was that we could decrease the DPI's maintaining the size in pixels, with the result of lose of quality but also decreasing the size of the image. I'm not speaking of changing the metadata, but the actual image resolution. Now I'm doing some research based in your answers and it seems only apply to printing, but on screen the only thing that matters is the size of the image, since decreasing dpi doesn't change the size of the file (as I thought), it's only an informational data. DPI's on digital is conditioned by the screen resolution, not dpi information on the file. Is that summary correct? Thank you very much for your time!

**Reply by NiklasBr** [reply]
> … we could decrease the DPI's maintaining the size in pixels …

This is correct.

> … with the result of lose of quality but also decreasing the size of the image

But this is not entirely so, unless by "size" you mean the physical size on paper or screen. It will have no effect on the number of bytes or pixels of the image file.

- DPI – (ink) Dots Per Inch
- Resolution of the image in pixels

These are distinct and independent of each other. A pixel does not have a physical size. For example: The resolution of a [jumbotron](https://en.wikipedia.org/wiki/Jumbotron) could be 1280 x 720 pixels but at a DPI of perhaps 2 pixels per inch. All while a mobile phone in landscape mode may also be 1280 x 720 pixels also, but at well over 250 pixels per inch.

You would use the same image file for both of them.

The calculation also works in reverse, say you want to print an image which is 4 inches across at 300 DPI, that would then require a 4 * 300 = 1200 pixel image. A 1500 pixel image printed at 300 DPI would be 1500 / 300 = 5 inches.

> … it's only an informational data

Yes, pretty much so. I guess the best course of action to determine the image sizes needed is to ask what the expected print sizes are (and the printer capability in terms of DPI), and then calculate the minimum and maximum image sizes to deliver from there.

**Reply by IMCtoys** [reply]
Understood! thank you very much for your time and sharing your knowledge!
